WPMedia/arc-themes-components
# Tmedia 640 component generator

Ticket: https://arcpublishing.atlassian.net/browse/TMEDIA-640

- Took the component taking in children but can remove the functionality if necessary 
- Didn't add any specific classnames based on reading the ticket
- Added placeholder style files for updating style functionality 

How to test: 

1. Install new hygen dep and update `rm -rf node_modules && npm i`
2. Run `npm run generate:component`
3. Enter word or phrase 'date area'
![Screen Shot 2022-01-24 at 13 56 31](https://user-images.githubusercontent.com/5950956/150855032-f047f99d-60ef-4c84-afba-7d2c20c7d87e.png)

4. See generated files 
![Screen Shot 2022-01-24 at 13 56 45](https://user-images.githubusercontent.com/5950956/150855068-267ecd9a-d920-4346-a887-b3169ecbc69f.png)
![Screen Shot 2022-01-24 at 13 57 01](https://user-images.githubusercontent.com/5950956/150855113-609aa457-1259-459f-9061-55d163a40bf2.png)

5. Run `npm run storybook`
<img width="403" alt="Screen Shot 2022-01-24 at 13 57 33" src="https://user-images.githubusercontent.com/5950956/150855190-7940fdc8-3098-4bf6-bcc1-a783d03e2cbd.png">

6. Run tests to validate `npm t`
![Screen Shot 2022-01-24 at 13 59 09](https://user-images.githubusercontent.com/5950956/150855424-c33d4484-9802-4ac5-b1d8-8498442aa164.png)

7. Run `npm run lint` to ensure that code style is followed for new component
AC: 

Implement a npm script - npm run generate:component that

Prompts the user to enter a component name, with the given component name the generator will (example adding a date component)

Create a folder with the component name (all lower case) in src/components → src/components/date

Create the following files in src/components/date

Each file created should have the boiler plate code to enable a developer to begin development

index.jsx

_index.scss

index.stories.mdx

index.test.jsx

Update the root index.js file to

add the import for the new component

export the component 

Update the root _index.scss file to @import the components scss file
